contrib/mesos/pkg/scheduler/podtask/predicate.go

```diff
@@ -57,12 +61,18 @@ func NodeSelectorPredicate(t *T, offer *mesos.Offer) bool {
 
 	// check the NodeSelector
 	if len(t.Pod.Spec.NodeSelector) > 0 {
-		slaveLabels := map[string]string{}
-		for _, a := range offer.Attributes {
-			if a.GetType() == mesos.Value_TEXT {
-				slaveLabels[a.GetName()] = a.GetText().GetValue()
+		slaveLabels := node.SlaveAttributesToLabels(offer.Attributes)
+
+		// add non-mesos labels
```

Comment:
We have to remove old labels. It can be that a slave doesn't have a label anymore it had before.